### PR TITLE
completed ticket 11

### PR DESCRIPTION
**Pull request by dmarkesini**
-used TDD
-error handled if the query topic does not exist in databse

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -275,6 +276,79 @@ describe("GET /api/articles", () => {
...
+      });
+  });
+  test("status: 404 for a topic that does not exist in the database", () => {
+    return request(app)
+      .get("/api/articles?topic=abc")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Topic not found!");
+      });
+  });
```
Comment (HeatherEver) [inline __tests__/app.test.js:350]: can you please add some tests for invalid sort_by and invalid order

File: __tests__/app.test.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const request = require("supertest");
 const db = require("../db/connection");
 const testData = require("../db/data/test-data/index");
 const seed = require("../db/seeds/seed");
+const { expect } = require("@jest/globals");
```
Comment (HeatherEver) [inline __tests__/app.test.js]: weird jest auto import - just remove this line in case it causes any issues

**Review by HeatherEver — COMMENTED** [review]
Lovely! Just a couple of tests to add in which case you might want to add a valid sort_by and valid order_by comparison so people can't add dodgy things into a query

File: __tests__/app.test.js (hunk 1 above)
Comment (Rosem10) [reply __tests__/app.test.js:350]: What if the topic exists but there are no associated artickes?

File: models/articles.models.js [hunk 3]
```diff
@@ -92,17 +92,50 @@ exports.selectCommentsById = (id) => {
...
+            )
+            .then(({ rows }) => {
+              return rows;
+            });
+        }
+      });
+  } else {
+    return db
+      .query(
+        `SELECT articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes,
```
Comment (Rosem10) [inline models/articles.models.js]: This works and is fine - but you could make it waaaay DRY-er by having one 'base' query string saved as a variable, and concatenating the values you want (e.g topic) to it, depending on whether there is one or not. At the moment there is a lot of crossover between the two db.queries.

**Review by Rosem10 — CHANGES_REQUESTED** [review]
Just one more test to add please!

**Review by HeatherEver — APPROVED** [review]

**Pull request merged**